src/script/event/EventRepository.ts

```diff
@@ -332,15 +334,14 @@ export class EventRepository {
    * @param source Source of notification
    */
   private async distributeEvent(event: IncomingEvent, source: EventSource) {
+    await Promise.all(this.eventProcessors.map(processor => processor.processEvent(event, source)));
```

Comment:
Now all processors will be awaited before continuing (will remove the `hack` on line 358 in this file). 
In the long run this should make things way cleaner

Reply:
**question**: Wouldn't it be more safe to use `Promise.allSettled` instead to be sure all of the processors fire? With `.all` it will reject after one of the processors fail.

Reply:
I assumed that we want all of them to succeed?

Reply:
Depends if we want to follow `all-or-nothing` or `at-least-something` approach 😅

Reply:
There's only one processor right now, there could be a lot of them in the future, I don't think one of them failing should abort all the rest (and then also the order matters, which is not good i'd say).

Reply:
Never mind, `.all` will not make all promises stop executing, it will just not await them (wait for their result), they should still be running in the background.